consensus/propagation/have_wants.go

```diff
@@ -254,8 +245,8 @@ func (blockProp *Reactor) handleRecoveryPart(peer p2p.ID, part *proptypes.Recove
 	// not this node must disconnect from them.
 	cb, parts, _, has := blockProp.getAllState(part.Height, part.Round, false)
 	if !has {
-		blockProp.Logger.Debug("received part for unknown proposal", "peer", peer, "height", part.Height, "round", part.Round)
-		// d.pswitch.StopPeerForError(p.peer, fmt.Errorf("received part for unknown proposal"))
+		blockProp.Logger.Error("received part for unknown proposal", "peer", peer, "height", part.Height, "round", part.Round)
+		blockProp.Switch.StopPeerForError(p.peer, errors.New("received recovery part for unknown proposal"))
 		return
 	}
```

Comment:
I don't think we should add this one yet due to pruning. this will definitely get hit under load if nodes are congested.

Reply:
05b0056448553ddddedf781aaf34074e3347f21d